integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/lru/CarbonDataShowLRUCommand.scala

```diff
@@ -0,0 +1,197 @@
+
+  override def output: Seq[Attribute] = {
+    Seq(AttributeReference("database", StringType, nullable = false)(),
+      AttributeReference("table", StringType, nullable = false)(),
+      AttributeReference("index size", LongType, nullable = false)(),
+      AttributeReference("dictionary size", LongType, nullable = false)())
+  }
+
+  override protected def opName: String = "SHOW LRU"
+
+  def showLRUOfAllTables(sparkSession: SparkSession): Seq[Row] = {
+    val currentDatabase = sparkSession.sessionState.catalog.getCurrentDatabase
+    val lruCache = CacheProvider.getInstance().getCarbonLRUCache()
+    if (lruCache == null) {
+      Seq(Row("ALL", "ALL", 0L, 0L),
+        Row(currentDatabase, "ALL", 0L, 0L))
+    } else {
+      val tableIdents = sparkSession.sessionState.catalog.listTables(currentDatabase).toArray
+      val dbLocation = CarbonEnv.getDatabaseLocation(currentDatabase, sparkSession)
+      val tempLocation = dbLocation.replace(
+        CarbonCommonConstants.WINDOWS_FILE_SEPARATOR, CarbonCommonConstants.FILE_SEPARATOR)
+      val tablePaths = tableIdents.map { tableIdent =>
+        (tempLocation + CarbonCommonConstants.FILE_SEPARATOR +
+         tableIdent.table + CarbonCommonConstants.FILE_SEPARATOR,
+          tableIdent.database.get + "." + tableIdent.table)
+      }
+
+      val dictIds = tableIdents.flatMap { tableIdent =>
+        CarbonEnv
+          .getCarbonTable(tableIdent)(sparkSession)
+          .getAllDimensions
+          .asScala
+          .filter(_.isGlobalDictionaryEncoding)
+          .toArray
+          .map(dim => (dim.getColumnId, tableIdent.database.get + "." + tableIdent.table))
+      }
+
+      // all databases
+      var (allIndexSize, allDictSize) = (0L, 0L)
+      // current database
+      var (dbIndexSize, dbDictSize) = (0L, 0L)
+      val tableMapIndexSize = mutable.HashMap[String, Long]()
+      val tableMapDictSize = mutable.HashMap[String, Long]()
+      val cacheIterator = lruCache.getLruCacheMap.entrySet().iterator()
+      while (cacheIterator.hasNext) {
+        val entry = cacheIterator.next()
+        val cache = entry.getValue
+        if (cache.isInstanceOf[BlockletDataMapIndexWrapper]) {
+          allIndexSize = allIndexSize + cache.getMemorySize
+          val indexPath = entry.getKey.replace(
+            CarbonCommonConstants.WINDOWS_FILE_SEPARATOR, CarbonCommonConstants.FILE_SEPARATOR)
+          val tablePath = tablePaths.find(path => indexPath.startsWith(path._1))
+          if (tablePath.isDefined) {
+            dbIndexSize = dbIndexSize + cache.getMemorySize
+            val memorySize = tableMapIndexSize.get(tablePath.get._2)
+            if (memorySize.isEmpty) {
+              tableMapIndexSize.put(tablePath.get._2, cache.getMemorySize)
+            } else {
+              tableMapIndexSize.put(tablePath.get._2, memorySize.get + cache.getMemorySize)
+            }
+          }
+        } else if (cache.isInstanceOf[AbstractColumnDictionaryInfo]) {
```

Comment:
Will this pr will support the BloomDataMapCache?

Reply:
Yes....it does support bloom